test-app/tests/integration/components/file-input-field-test.gts

```diff
@@ -0,0 +1,274 @@
+    // wrapping <label> element
+    let describedby =
+      find('[data-file-input-field]')?.getAttribute('aria-describedby') || '';
+    assert.ok(
+      describedby.includes(errorId),
+      'Expected errorId to be included in the aria-describedby'
+    );
+
+    assert.dom('[data-file-input-field]').hasAttribute('aria-invalid', 'true');
+
+    assert.dom('[data-file-input-field]').hasClass('shadow-error-outline');
+    assert
+      .dom('[data-file-input-field]')
+      .doesNotHaveClass('shadow-focusable-outline');
+  });
+
+
+  test('it sets the "for" attribute on the label to the "id" attribute of the file input field', async function (assert) {
+    await render(<template>
+      <FileInputField @label="Label" data-file-input-field />
+    </template>);
+
+    let labelFor = find('label')?.getAttribute('for') || '';
+    assert.ok(labelFor, 'Expected the id attribute of the label to be truthy');
+
+    assert
+      .dom('[data-file-input-field]')
+      .hasAttribute(
+        'id',
+        labelFor,
+        'Expected the for attribute on the label to match the id attribute on the checkbox'
+      );
+  });
+
+  test('it disables the file input using @isDisabled', async function (assert) {
+    await render(<template>
+      <FileInputField
+        @label="Label"
+        @isDisabled={{true}}
+        data-file-input-field
+      />
+    </template>);
+
+    assert.dom('[data-file-input-field]').isDisabled();
+    assert.dom('[data-file-input-field]').hasClass('text-disabled');
+  });
+ 
+  test('it spreads attributes to the underlying input', async function (assert) {
+    await render(<template>
+      <FileInputField
+        @label="Label"
+        placeholder="Placeholder text"
+        data-file-input-field
+      />
+    </template>);
+
+    assert
+      .dom('[data-file-input-field]')
+      .hasAttribute('placeholder', 'Placeholder text');
+  });
+
+  test('it can upload a file and display the filename', async function (assert) {
+    await render(<template>
+      <FileInputField @label="Label" data-file-input-field />
+    </template>);
+
+    const file = createFile(['Upload file sample'], {
+      name: 'sample.txt',
+      type: 'text/plain',
+    });
+
+    triggerEvent('[data-file-input-field]', 'change', { files: [file] });
+
+    await settled();
+
+    assert.dom('[data-files] [data-file-name]').hasText('sample.txt');
+
+    // currently: rounding to the nearest KB
+    // this file is very small so result is 0 KB
+    assert.dom('[data-files] [data-file-size]').hasText('0 KB');
+  });
+
+  test('it calls @onChange when input is received', async function (assert) {
+    assert.expect(6);
+
+    const handleChange = (files: File[], e: Event | InputEvent) => {
+      assert.ok(e, 'Expected `e` to be available as the second argument');
+      assert.ok(e.target, 'Expected direct access to target from `e`');
+      assert.strictEqual(files.length, 1, 'Expected a single file to be uploaded');
```

Comment:
Can we check more that just that this is an array? Like `files[0] instanceof File`, and maybe the name of it?